File: C++/source/Framework/Messaging/LightHouse/LightPulseDispatchMessage.cpp
/***************************************************************************
 *   Copyright (C) 20016 Vidyasagara Guntaka                               *
 *   All rights reserved.                                                  *
 *   Author : Vidyasagara Reddy Guntaka                                    *
 ***************************************************************************/

#include "Framework/Messaging/LightHouse/LightPulseDispatchMessage.h"
#include "Framework/Messaging/LightHouse/LightPulseDispatchObjectManager.h"
#include "LightPulseDispatchTypes.h"

namespace WaveNs
{

LightPulseDispatchMessage::LightPulseDispatchMessage ()
    : WaveMessage (LightPulseDispatchObjectManager::getWaveServiceId (), LIGHT_PULSE_DISPATCH_RECEIVED_LIGHT_PULSE)
{
}

LightPulseDispatchMessage::~LightPulseDispatchMessage ()
{
}

string LightPulseDispatchMessage::getLightPulseSerializedString () const
{
    return (m_lightPulseSerializedString);
}

void LightPulseDispatchMessage::setLightPulseSerializedString (const string &lightPulseSerializedString)
{
    m_lightPulseSerializedString = lightPulseSerializedString;
}

void LightPulseDispatchMessage::setupAttributesForSerialization ()
{
    WaveMessage::setupAttributesForSerialization ();

    addSerializableAttribute (new AttributeString (&m_lightPulseSerializedString, "lightPulseSerializedString"));
}

string LightPulseDispatchMessage::getSenderIpAddress () const
{
    return (m_senderIpAddress);
}

void LightPulseDispatchMessage::setSenderIpAddress (const string &senderIpAddress)
{
    m_senderIpAddress = senderIpAddress;
}

SI32 LightPulseDispatchMessage::getSenderPort () const
{
    return (m_senderPort);
}

void LightPulseDispatchMessage::setSenderPort (const SI32 &senderPort)
{
    m_senderPort = senderPort;
}

}

File: Java/source/com/CxWave/Wave/Framework/Messaging/LightHouse/LightPulseDispatchMessage.java
/*************************************************************************************************************************
 * Copyright (C) 2015-2016 Vidyasagara Guntaka & CxWave, Inc * All rights reserved. * Author : Vidyasagara Reddy Guntaka *
 *************************************************************************************************************************/

package com.CxWave.Wave.Framework.Messaging.LightHouse;

import com.CxWave.Wave.Framework.Messaging.Local.WaveMessage;
import com.CxWave.Wave.Framework.Type.SI32;

public class LightPulseDispatchMessage extends WaveMessage
{
    private String m_lightPulseSerializedString;
    private String m_senderIpAddress;
    private SI32   m_senderPort = new SI32 (0);

    public LightPulseDispatchMessage ()
    {
        super (LightPulseDispatchObjectManager.getWaveServiceId (), LightHouseDispatchTypes.LIGHT_PULSE_DISPATCH_RECEIVED_LIGHT_PULSE);
    }

    public LightPulseDispatchMessage (final String lightPulseSerializedString)
    {
        super (LightPulseDispatchObjectManager.getWaveServiceId (), LightHouseDispatchTypes.LIGHT_PULSE_DISPATCH_RECEIVED_LIGHT_PULSE);

        m_lightPulseSerializedString = lightPulseSerializedString;
    }

    public String getLightPulseSerializedString ()
    {
        return m_lightPulseSerializedString;
    }

    public void setLightPulseSerializedString (final String lightPulseSerializedString)
    {
        m_lightPulseSerializedString = lightPulseSerializedString;
    }

    public String getSenderIpAddress ()
    {
        return m_senderIpAddress;
    }

    public void setSenderIpAddress (final String senderIpAddress)
    {
        m_senderIpAddress = senderIpAddress;
    }

    public SI32 getSenderPort ()
    {
        return m_senderPort;
    }

    public void setSenderPort (final SI32 senderPort)
    {
        m_senderPort = senderPort;
    }
}
